src/media/js/elements/nav.js

```diff
@@ -0,0 +1,149 @@
+            subnavs: {
+                get: function() {
+                    return $(this).children('mkt-nav-child');
+                },
+            },
+        }),
+    });
+
+    var MktNavRootElement = document.registerElement('mkt-nav-root', {
+        prototype: Object.create(HTMLUListElement.prototype, {}),
+    });
+
+    var MktNavChildElement = document.registerElement('mkt-nav-child', {
+        prototype: Object.create(MktNavRootElement.prototype, {
+
+            // Show this subnav and hide all sibling <mkt-nav-child> elements.
+            show: {
+                value: function() {
+                    var $this = $(this);
+                    $this.siblings('.' + VISIBLE).removeClass(VISIBLE);
+                    $this.addClass(VISIBLE);
+                    return this;
+                },
+            }
+        }),
+    });
+
+    // Toggle the <mkt-nav> element with the specified id when elements with the
+    // data-toggle-nav attribute are clicked.
+    z.body.on('click', '[data-toggle-nav]', function(evt) {
+        evt.preventDefault();
+        evt.stopPropagation();
+        var navId = $(this).data('toggle-nav');
+        $('#' + navId).get(0).toggle();
+    })
+
+    // Toggle the <mkt-nav-child> element with the specified id when elements
+    // with the data-toggle-nav attribute are clicked. If the value of that
+    // attribute is empty, attempt to toggle the parent's subnav if it is an
+    // <mkt-nav> element.
+    .on('click', '[data-toggle-subnav]', function(evt) {
+        evt.preventDefault();
+        evt.stopPropagation();
+        var $this = $(this);
+        var id = $this.data('toggle-subnav');
+        if (id) {
+            var $subnav = $('#' + id);
+            if ($subnav.is('.' + VISIBLE)) {
+                $subnav.parent().get(0).toggleSubnav();
+                $this.removeClass(SHOWING);
+            } else {
+                $subnav.get(0).show();
+                $subnav.parent().get(0).toggle(true).toggleSubnav(true);
+                $this.addClass(SHOWING);
+            }
+        } else {
+            var $parent = $this.parent();
+            if ($parent.is('mkt-nav')){
+                $parent.get(0).toggleSubnav();
+            }
+        }
+    });
+
+    $(function() {
+
+    })
+
+    z.page.on('navigate', function(evt) {
+        // Remove highlights from formerly-active nodes.
+        $('mkt-nav a.' + ACTIVE).each(function(index, element) {
+            $(element).removeClass(ACTIVE);
+        });
+
+        // Highlight new active nodes.
+        $('mkt-nav a[href="' + window.location.pathname + '"]').each(function(index, element) {
+            $(element).addClass(ACTIVE);
+        });
+
+        // Close all menus.
+        $('mkt-nav').each(function(index, element) {
+            if (element.hasOwnProperty('toggle')) {
+                element.toggle(false).toggleSubnav(false); 
+            }
+        });
+
+        $('.' + SHOWING).removeClass(SHOWING);
```

Comment:
Yeah, we could, and I originally did. But there were some places where it as a lot more awkward to use DOM directly than jQuery ([e.g.](https://github.com/mozilla/marketplace-style-guide/pull/11/files#diff-6b10b5da552b0a923f7fa0bbdaf1f35bR74)), and I didn't want to mix jQuery with raw DOM selection in the same module.